### PR TITLE
Make sync API consistent: always (before, after)

### LINKED ISSUE #1293: Use consistent parameter order of (before, after) throughout update APIs
Currently most of the update APIs use the parameter order (before, after), but there are some that have the order (after, before). Since these parameters are the same type the compiler won't be able to catch it if we accidentally call them with (before, after) and we'll get runtime bugs where an update gets applied in the "wrong direction". We should make the update APIs consistent so that any APIs that take (before, after) or (old, new) parameters always take them in the same order.

### PR DESCRIPTION
**Pull request by rmunn**
Also update all call sites so that code behavior remains the same.

Fix #1293.

### CODE REVIEW

File: backend/FwLite/MiniLcm/SyncHelpers/WritingSystemSync.cs [hunk 1]
```diff
@@ -5,15 +5,15 @@ namespace MiniLcm.SyncHelpers;
 
 public static class WritingSystemSync
 {
-    public static async Task<int> Sync(WritingSystems currentWritingSystems,
-        WritingSystems previousWritingSystems,
+    public static async Task<int> Sync(WritingSystems previousWritingSystems,
```
Comment (hahn-kev) [inline backend/FwLite/MiniLcm/SyncHelpers/WritingSystemSync.cs:8]: while we're at it lets change these from `previous` to before, and `current` to after for all the WS sync code

File: backend/FwLite/MiniLcm/SyncHelpers/SemanticDomainSync.cs [hunk 2]
```diff
@@ -5,8 +5,8 @@ namespace MiniLcm.SyncHelpers;
 
 public static class SemanticDomainSync
 {
-    public static async Task<int> Sync(SemanticDomain[] currentSemanticDomains,
-        SemanticDomain[] previousSemanticDomains,
+    public static async Task<int> Sync(SemanticDomain[] previousSemanticDomains,
```
Comment (hahn-kev) [inline backend/FwLite/MiniLcm/SyncHelpers/SemanticDomainSync.cs:8]: again lets change this to before after

File: backend/FwLite/MiniLcm/SyncHelpers/PartOfSpeechSync.cs [hunk 3]
```diff
@@ -5,8 +5,8 @@ namespace MiniLcm.SyncHelpers;
 
 public static class PartOfSpeechSync
 {
-    public static async Task<int> Sync(PartOfSpeech[] currentPartsOfSpeech,
-        PartOfSpeech[] previousPartsOfSpeech,
+    public static async Task<int> Sync(PartOfSpeech[] previousPartsOfSpeech,
```
Comment (hahn-kev) [inline backend/FwLite/MiniLcm/SyncHelpers/PartOfSpeechSync.cs:8]: here too

**Review by hahn-kev — CHANGES_REQUESTED** [review]
looks good, let's change the current previous terminology that's still in some of the sync code at the same time to stay consistent

File: backend/FwLite/MiniLcm/SyncHelpers/WritingSystemSync.cs (hunk 1 above)
Comment (rmunn) [reply backend/FwLite/MiniLcm/SyncHelpers/WritingSystemSync.cs]: Done in commit af5dfe7c.

File: backend/FwLite/MiniLcm/SyncHelpers/SemanticDomainSync.cs (hunk 2 above)
Comment (rmunn) [reply backend/FwLite/MiniLcm/SyncHelpers/SemanticDomainSync.cs]: Done in commit af5dfe7c.

File: backend/FwLite/MiniLcm/SyncHelpers/PartOfSpeechSync.cs (hunk 3 above)
Comment (rmunn) [reply backend/FwLite/MiniLcm/SyncHelpers/PartOfSpeechSync.cs]: Done in commit af5dfe7c.

**Comment by rmunn** [comment]
Whoops. Didn't see that #1267 hasn't been merged yet. Closing this one, as merging it before merging #1267 risks ending up with bugs as a result of merge resolution.

**Pull request closed without merging**